external/storm-kafka-client/src/main/java/org/apache/storm/kafka/spout/internal/fetcher/KafkaRecordsFetchers.java

```diff
@@ -0,0 +1,41 @@
+package org.apache.storm.kafka.spout.internal.fetcher;
+
+import org.apache.kafka.clients.consumer.ConsumerRebalanceListener;
+import org.apache.kafka.clients.consumer.KafkaConsumer;
+import org.apache.storm.kafka.spout.KafkaPartitionReader;
+import org.apache.storm.kafka.spout.KafkaPartitionReaders;
+import org.apache.storm.kafka.spout.KafkaSpout;
+import org.apache.storm.kafka.spout.KafkaSpoutConfig;
+import org.apache.storm.task.TopologyContext;
+
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Created by liurenjie on 19/12/2016.
+ */
+public final class KafkaRecordsFetchers {
+    public static <K, V> KafkaRecordsFetcher<K, V> kafkaRecordsFetcherOf(KafkaSpoutConfig kafkaSpoutConfig,
+                                                                         KafkaConsumer<K, V> consumer,
+                                                                         TopologyContext context,
+                                                                         ConsumerRebalanceListener rebalanceListener) {
+        if (kafkaSpoutConfig.getManualPartitionAssignment()) {
```

Comment:
Agree. If you need to reuse some code across the manual and automatic mode, put it in a base abstract class, and then provide two implementations, one for auto and another for manual.